src/wedding.cpp

```diff
@@ -0,0 +1,399 @@
+
+/* $Id$
+ * EOSERV is released under the zlib license.
+ * See LICENSE.txt for more info.
+ */
+
+#include "wedding.hpp"
+
+#include "character.hpp"
+#include "map.hpp"
+#include "npc.hpp"
+#include "packet.hpp"
+#include "timer.hpp"
+#include "world.hpp"
+
+#include "console.hpp"
+#include "util.hpp"
+
+#include <string>
+
+static void wedding_tick(void *wedding_void)
+{
+	Wedding* wedding = static_cast<Wedding*>(wedding_void);
+
+	wedding->Tick();
+}
+
+NPC* Wedding::GetPriest()
+{
+	return this->map->GetNPCIndex(this->priest_idx);
+}
+
+Character* Wedding::GetPartner1()
+{
+	return this->map->GetCharacter(this->partner1);
+}
+
+Character* Wedding::GetPartner2()
+{
+	return this->map->GetCharacter(this->partner2);
+}
+
+void Wedding::PriestSay(const std::string& message)
+{
+	NPC* priest = this->GetPriest();
+
+	if (priest)
+		priest->Say(message);
+}
+
+void Wedding::StartTimer()
+{
+	if (!this->tick_timer)
+	{
+		this->tick_timer = new TimeEvent(wedding_tick, this, 1.5, Timer::FOREVER);
+		this->map->world->timer.Register(this->tick_timer);
+	}
+}
+
+void Wedding::StopTimer()
+{
+	if (this->tick_timer)
+	{
+		this->map->world->timer.Unregister(this->tick_timer);
```

Comment:
`Unregister` doesn't appear to delete the timer that gets passed to it. Does this get leaked?

Reply:
Looks like it does, good catch.
Easy fix, simply deleting `tick_timer` should do the trick. A registered `TimeEvent` will unregister itself in the destructor.